app/controllers/chat_channel_memberships_controller.rb

```diff
@@ -115,14 +115,49 @@ def leave_membership
     end
   end
 
+  def update_membership_role
+    @chat_channel = ChatChannel.find_by(id: params[:id])
+    authorize @chat_channel, :update?
+    membership = ChatChannelMembership.find_by(id: channel_membership_params[:membership_id], chat_channel_id: @chat_channel.id)
+
+    membership.update(role: channel_membership_params[:role])
+    if membership.errors.any?
+      render json: { success: false, message: "Failed to update membership", errors: chat_channel_membership.errors.full_messages }, status: :bad_request
+    else
+      render json: { success: true, message: "User Membership is updated" }, status: :ok
+    end
+  end
+
+  def join_channel_invitation
```

Comment:
As I understand this, since it relies on a channel slug there's no real way to "expire" the link.

Could we instead have something that could expire?

Also I feel like just _automatically joining a channel_ based on a link you could click on accidentally is not ideal. What if you needed to land on a page first and then proactively click a button that joined via a `POST` request?

Reply:
@benhalpern 
- Yes, we can expire the link.
- Also, I will add the page that will ask user to `join the channel` or `cancel`, when user use the joining link.

quick question: 
- What is the expiry time for the link?  => (1day, 6hours)

Reply:
I think a one day expiration makes sense

Reply:
@benhalpern added expiry for an invitation link